### PR TITLE
boards: atsame54_xpro: Add SAM E54 Xplained Pro Evaluation Kit

### PR DESCRIPTION
**Pull request by benpicco**
This adds support for the Atmel SAM E54 Xplained Pro Evaluation Kit.

split off from #14685
depends on ~~#20062~~  ~~#20056~~ ~~#20058~~
additional functionality depends on ~~#20052~~ ~~#20055~~ ~~#20057~~ ~~#21866~~

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
doc change LGTM, thanks.

**Review by galak — CHANGES_REQUESTED** [review]
Can you shrink the images, they are huge (file size).

Does this only need the SoC PR, or do we need UART driver?  GPIO driver?  [ what's the minimal set of SoC + drivers to make the board code useful ].

**Comment by benpicco** [comment]
Down to
```
808K ATSAME54-XPRO-pinout.png 
596K atsame54_xpro.png
```

If you just want to blink an LED, the SoC PR is enough.
The GPIO PR is about EXTI (and port D), basic GPIO functionality is the same as on samd2x.

UART would be nice of course. :wink:

**Comment by galak** [comment]
> Down to
> 
> ```
> 808K ATSAME54-XPRO-pinout.png 
> 596K atsame54_xpro.png
> ```
> 
> If you just want to blink an LED, the SoC PR is enough.
> The GPIO PR is about EXTI (and port D), basic GPIO functionality is the same as on samd2x.
> 
> UART would be nice of course. 😉

Sent you on slack a much smaller image, take a look and see what you think.

**Comment by galak** [comment]
@benpicco can you look at addressing the build issues / sanitycheck failures.  If you want this in for 2.1 we need to resolve this by friday (Nov 8th).

**Comment by benpicco** [comment]
I forgot to also move the `flash` and `sram` blocks in the `samd5xx20.dtsi` etc DTS files after moving it in `samd5x.dts` - this is fixed by #20420 

Then the DTS processor will complain about the unknown lock regions property introduced by #20058 - you'll get past the config with this PR 😉 

Then `soc.h` wants to include `tc_fixup_samd5x.h` which is introduced by #20056

With these three, `hello_world` compiles.

**Comment by ioannisg** [comment]
> This adds support for the Atmel SAM E54 Xplained Pro Evaluation Kit.
> 
> split off from #14685
> depends on #20062

#20062 is merged; update the description :)

File: boards/arm/atsame54_xpro/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Kconfig - SAM E54 Xplained Pro Board configuration
```
Comment (ioannisg) [inline boards/arm/atsame54_xpro/Kconfig.board]: Remove Kconfig, start with "SAM E54 ..."

Comment (ioannisg) [reply boards/arm/atsame54_xpro/Kconfig.board]: And the same for all these license headers :)

File: tests/drivers/spi/spi_loopback/boards/atsame54_xpro.conf [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_0"
```
Comment (ioannisg) [inline tests/drivers/spi/spi_loopback/boards/atsame54_xpro.conf]: Why is this change here?

**Review by ioannisg — CHANGES_REQUESTED** [review]
The test file looks not relevant to the PR.
Some clean up of the Kconfig file headers is needed.
Looking good, otherwise

File: tests/drivers/spi/spi_loopback/boards/atsame54_xpro.conf (hunk 2 above)
Comment (benpicco) [reply tests/drivers/spi/spi_loopback/boards/atsame54_xpro.conf]: So you can run the `spi_loopback` test on this board - other boards [have it too](https://github.com/zephyrproject-rtos/zephyr/tree/master/tests/drivers/spi/spi_loopback/boards/)

Comment (ioannisg) [reply tests/drivers/spi/spi_loopback/boards/atsame54_xpro.conf]: Sure, I was wondering "why part of this PR" - is the SPI driver already merged?

Comment (benpicco) [reply tests/drivers/spi/spi_loopback/boards/atsame54_xpro.conf]: Alright I moved it to the SPI PR.

**Review by ioannisg — APPROVED** [review]

**Comment by galak** [comment]
@benpicco still major CI failures

**Comment by benpicco** [comment]
@galak I removed references the not yet merged features.

**Comment by benpicco** [comment]
If the other PRs get merged before this, I will remove the last few commits here.

**Comment by ioannisg** [comment]
@benpicco can you address the checkpatch warnings?

**Comment by benpicco** [comment]
@ioannisg I'm not sure how - I can't find documentation for the vendor strings in [`atsamd21_xpro.dts`](https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/atsamd21_xpro/atsamd21_xpro.dts#L12) either - where should they go?

**Comment by ioannisg** [comment]
@galak can you take a look?

**Review by galak — APPROVED** [review]

**Comment by benpicco** [comment]
Thank you for the review & merge!
